layernom inplace

Krovatkin opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63437
* #63436

Differential Revision: [D30388824](https://our.internmc.facebook.com/intern/diff/D30388824)

Krovatkin committed:
layernom inplace

[ghstack-poisoned]

eellison commented on test/jit/test_freezing.py:
```diff
@@ -1939,7 +1950,42 @@ def test_layernorm(self):
+                            inplace_suffix = "_" if inplace else ""
+                            FileCheck().check("prim::MKLDNNLayerNorm" + inplace_suffix).check_count("aten::to_dense", 1, exactly=True).run(mod.graph)
+                        else:
+                            FileCheck().check_count("aten::to_dense", 1, exactly=True).run(mod.graph)
```
this is not checking anything, we should be checking that aten::layernorm exists or whatever

eellison commented on test/jit/test_freezing.py:
```diff
@@ -1939,7 +1950,42 @@ def test_layernorm(self):
+    @skipIfNoTorchVision
+    def test_layernorm2(self):
+        with set_default_dtype(torch.float):
+            model = torchvision.models.resnet18()
```
model is unused

eellison commented on test/jit/test_freezing.py:
```diff
@@ -1939,7 +1950,42 @@ def test_layernorm(self):
+            N, C, H, W, = 10, 3, 224, 224
+            for param in ((torch.nn.Linear(H, W), [W], torch.randn(N, C, W)),):
+
+                for layernorm in (torch.nn.LayerNorm(param[1]),):
+                    sub_model = ResidualLayernorm(param[0], layernorm)
```
why do we need this test in addition to what exists above ?

eellison commented on test/jit/test_freezing.py:
```diff
@@ -1939,7 +1950,42 @@ def test_layernorm(self):
+                        self.assertTrue(torch.allclose(sub_model(param[2]), mod(param[2]), 1e-04, 1e-04))
+
+    @skipIfNoTorchVision
+    def test_layernorm2(self):
```
can you have a more descriptive testname ? what is this testing that the other test is not

Krovatkin committed:
Update on "layernom inplace"




[ghstack-poisoned]

Krovatkin commented:
@krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30388824).

Krovatkin committed:
Update on "layernom inplace"


Differential Revision: [D30388824](https://our.internmc.facebook.com/intern/diff/D30388824)

[ghstack-poisoned]

Krovatkin commented:
@krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30388824).

Krovatkin committed:
Update on "layernom inplace"


Differential Revision: [D30388824](https://our.internmc.facebook.com/intern/diff/D30388824)

[ghstack-poisoned]

Krovatkin commented:
@krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30388824).
